packages/frontend/core/src/modules/tag/service.ts

```diff
@@ -0,0 +1,34 @@
+import type { Workspace } from '@toeverything/infra';
+
+import type { WorkspaceLegacyProperties } from '../workspace';
+import { Tag } from './entities/tag';
+
+export class TagService {
+  constructor(
+    private readonly properties: WorkspaceLegacyProperties,
+    private readonly workspace: Workspace
+  ) {}
+
+  readonly tags = this.properties.tagOptions$;
```

Comment:
services should return entity instance

```suggestion
  readonly tags = this.properties.tagOptions$.map(tags => 
    tags.map(
      tag => new Tag(tag.id, tag.name, this.properties, this.workspace)
    )
  );
```